Reflection/Login.simba

```diff
@@ -0,0 +1,299 @@
+  if Widget.ref <> nil then
+  begin
+    Mouse.Click(Widget.Bounds, MOUSE_LEFT);
+    Result := R_WaitUntilVisible(R_WORLD_SWITCHER_SCROLL_UP, 2500);
+    Widget.Free;
+  end;
+end;
+
+{*
+R_WorldToBox
+~~~~~~~~~~~~~~~~~~~~~~
+Utility function that converts a given world to mainscreen coordinates.
+
+Example:
+    R_WorldToBox(500);
+*}
+Function R_WorldToBox(World: Int32): TBox;
+var
+  WorldWidget: RSWidget;
+begin
+  WorldWidget := RSWidget.Get(R_WORLD_SWITCHER_LIST.Group, R_WORLD_SWITCHER_LIST.Child, World);
+  Result := WorldWidget.Bounds;
+  WorldWidget.Free;
+end;
+
+{*
+R_ScrollToWorld
+~~~~~~~~~~~~~~~~~~~~~~
+Scrolls to the proper world in the world switcher.
+
+Example:
+    R_ScrollToWorld(500, TRSWorldScroll.SCROLLBAR);
+*}
+Function R_ScrollToWorld(World: Int32; ScrollType: TRSWorldScroll = TRSWorldScroll.SCROLLWHEEL): Boolean;
+var
+  ScrollBar, ScrollArrow, WorldContainer: RSWidget;
+  WorldContainerBounds: TBox;
+  T: Timer;
+  Direction: Boolean;
+  P: TPoint;
+begin
+  WorldContainer := RSWidget.Get(R_WORLD_SWITCHER_LIST_INTERFACE_CONTAINER_CHILD.Group, R_WORLD_SWITCHER_LIST_INTERFACE_CONTAINER_CHILD.Child);
+  if WorldContainer.ref = nil then
+    Exit(False);
+
+  //Adds a buffer to make sure that the world is completely clickable
+  WorldContainerBounds := [WorldContainer.Bounds.X1, WorldContainer.Bounds.Y1 + 15, WorldContainer.Bounds.X2, WorldContainer.Bounds.Y2 - 15];
+  WorldContainer.Free;
+
+  if WorldContainerBounds.Contains(R_WorldToBox(World)) then
+    Exit(True);
+
+  Direction := WorldContainerBounds.Middle.Y > R_WorldToBox(World).Middle.Y;
+
+  //Scroll the interface using the mouse-wheel
+  if ScrollType = TRSWorldScroll.SCROLLWHEEL then
+  begin
+    Mouse.Move(WorldContainerBounds);
+
+    T.Start;
+    while not WorldContainerBounds.Contains(R_WorldToBox(World)) do
+    begin
+      if T.ElapsedTime > 5000 then
+        Exit(False);
+
+      Mouse.Scroll(3, Direction);
+      writeln(R_WorldToBox(World));
+    end;
+
+    Exit(WorldContainerBounds.Contains(R_WorldToBox(World).Middle));
+  end;
+
+  //Scroll the interface using the scroll-bar
+  if ScrollType = TRSWorldScroll.SCROLLBAR then
+  begin
+    ScrollBar := RSWidget.Get(R_WORLD_SWITCHER_SCROLL_BAR.Group, R_WORLD_SWITCHER_SCROLL_BAR.Child, R_WORLD_SWITCHER_SCROLL_BAR.Index);
+    if ScrollBar.ref = nil then
+      Exit(False);
+
+    Mouse.Move(ScrollBar.Bounds);
+    Mouse.Click(MOUSE_LEFT);
+    Mouse.Hold(MOUSE_LEFT);
+    ScrollBar.Free;
+
+    T.Start;
+    while not WorldContainerBounds.Contains(R_WorldToBox(World)) do
+    begin
+      if T.ElapsedTime > 5000 then
+      begin
+        Mouse.Release(MOUSE_LEFT);
+        Exit(False);
+      end;
+
+      P := Mouse.GetPosition;
+      if Direction then
+        Mouse.Move(P.X + RandomRange(-1, 1), P.Y - RandomRange(3, 7))
+      else
+        Mouse.Move(P.X + RandomRange(-1, 1), P.Y + RandomRange(3, 7));
```

Comment:
just a nit about randomness (obviously theres no proof they even look for this): this is good, but the X tol is a little tight imo

Reply:
Yea, I had chosen that range since it recalculates those positions every iteration, so I'd rather have the x direction move 1pixel at maximum per iteration than stuttering all over the place multiple times a second. I felt it gave it a bit more of a human-like movement, but maybe I am incorrect.

Reply:
ah I misunderstood the flow, you're right -- too much and it would look kinda weird. I'm fine with leaving it as is.